Add tests for liking and unliking posts

goobertron opened the pull request:
I noticed there were no tests in the desktop suite for 'like' interactions. There are now.

Commit:
Add tests for liking and unliking posts

goobertron commented:
Unh, these passed locally. Will get on to it.

svbergerem commented on features/desktop/likes.feature:
```diff
@@ -0,0 +1,37 @@
+    When I like the post "I like unicorns" in the stream
+    Then I should see "1 Like" within "#main_stream .stream_element .likes"
+
+	When I unlike the post "I like unicorns" in the stream
```
indentation

goobertron replied:
This is a weird one, because the indentation is fine in the file I pushed. Not sure why it has happened.

jhass replied:
It's a tab character, your editor most likely displays it the same.

goobertron replied:
Aha! Thanks, will make sure there are no tabs.

svbergerem commented on features/support/publishing_cuke_helpers.rb:
```diff
@@ -88,15 +88,21 @@ def find_post_by_text(text)
-  def within_post(post_text)
-    within find_post_by_text(post_text) do
-      yield
+  def like_show_page_post(post_text)
```
On the SPV there is no need to specify the post text

svbergerem commented on features/support/publishing_cuke_helpers.rb:
```diff
@@ -88,15 +88,21 @@ def find_post_by_text(text)
-    within find_post_by_text(post_text) do
-      yield
+  def like_show_page_post(post_text)
+    with_scope(post_text) do
```
Instead you could have `within("#single-post-actions") do` here

svbergerem commented on features/step_definitions/single_post_view_steps.rb:
```diff
@@ -12,4 +12,9 @@
+end
+
+And /^I click to (?:like|unlike) the post/ do
+  find('.like').click
```
You already defined `like_show_page_post` so you could use it here.

svbergerem commented on features/desktop/likes.feature:
```diff
@@ -0,0 +1,37 @@
+    Then I should see "1 Like" within "#main_stream .stream_element .likes"
+
+	When I unlike the post "I like unicorns" in the stream
+    Then I should not see "1 Like" within "#main_stream .stream_element .likes"
```
That is the only failing line because after unliking the post cucumber isn't able to find `.likes` anymore so the correct step would be
`Then I should not see a ".likes" within "#main_stream .stream_element"`

goobertron commented:
Have updated and rebased. I changed all the 'should see' lines to look for `.likes` or `.count` CSS, so this will be consistent with making tests text-independent. Hope it's all ok (cuke runs fine locally).

svbergerem commented:
Thank you!

Flaburgan commented:
Thanks @goobertron :)
